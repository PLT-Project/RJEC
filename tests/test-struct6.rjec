struct my_struct {
    a int;
    b char;
    c bool;
}

func foo(i struct my_struct, x int, y char) {
    i.a = x;
    i.b = y;

    printi(i.a);
    printc(i.b);
    printb(i.c);
}

func main() {

    j := struct my_struct {
        a : 10,
        b : 'r'
    };

    printi(j.a);
    printc(j.b);
    printb(j.c);

    foo(j, 20, 'j');

    printi(j.a);
    printc(j.b);
    printb(j.c);

}