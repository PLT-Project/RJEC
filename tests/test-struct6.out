10
r
false
20
j
false
10
r
false